backend/api/Repositories/Cdogs/CdogsAuthRepository.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Pims.Api.Helpers.Exceptions;
+using Pims.Api.Models;
+using Pims.Api.Models.Cdogs;
+
+namespace Pims.Api.Repositories.Cdogs
+{
+    /// <summary>
+    /// CdogsAuthRepository provides authentication to the Cdogs external api.
+    /// </summary>
+    public class CdogsAuthRepository : CdogsBaseRepository, IDocumentGenerationAuthRepository
+    {
+        private JwtResponse _currentToken;
+        private DateTime _lastSucessfullRequest;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CdogsAuthRepository"/> class.
+        /// </summary>
+        /// <param name="logger">Injected Logger Provider.</param>
+        /// <param name="httpClientFactory">Injected Httpclient factory.</param>
+        /// <param name="configuration">The injected configuration provider.</param>
+        public CdogsAuthRepository(
+            ILogger<CdogsAuthRepository> logger,
+            IHttpClientFactory httpClientFactory,
+            IConfiguration configuration)
+            : base(logger, httpClientFactory, configuration)
+        {
+            _currentToken = null;
+            _lastSucessfullRequest = DateTime.UnixEpoch;
+        }
+
+        public async Task<string> GetTokenAsync()
+        {
+            if (!IsValidToken())
+            {
+                ExternalResult<JwtResponse> tokenResult = await RequestToken();
+
+                if (tokenResult.Status == ExternalResultStatus.Error)
+                {
+                    throw new AuthenticationException(tokenResult.Message);
+                }
+
+                _lastSucessfullRequest = DateTime.UtcNow;
+                _currentToken = tokenResult.Payload;
+            }
+
+            return _currentToken.AccessToken;
+        }
+
+        private bool IsValidToken()
```

Comment:
Hmm, I don't see anything about the refresh token in this logic.

Reply:
There is no refresh token when using the client credentials grant

Reply:
oh, I saw a refresh token in the model that you created, so that is not populated?

Reply:
I based the JTW model on a general jtw response. If at some point we change the grant type it could be populated. I could remove the unused fields for clarity